scripts/build_osm_network.py

```diff
@@ -595,131 +595,97 @@ def create_station_at_equal_bus_locations(
     return lines, buses
 
 
-def _split_linestring_by_point(linestring, points):
+def fix_overpassing_lines(lines, buses, distance_crs, tol=1):
     """
-    Function to split a linestring geometry by multiple inner points.
+    Function to snap buses to lines that are within a certain tolerance.
 
     Parameters
     ----------
-    lstring : LineString
-        Linestring of the line to be split
-    points : list
-        List of points to split the linestring
-
-    Return
-    ------
-    list_lines : list
-        List of linestring to split the line
+    lines : GeoDataFrame
+        GeoDataFrame containing the lines
+    buses : GeoDataFrame
+        GeoDataFrame containing the buses
+    distance_crs : str
+        Coordinate reference system to use for distance calculations
+    tol : float
+        Tolerance in meters to snap the buses to the lines
+
+    Returns
+    -------
+    lines : GeoDataFrame
+        GeoDataFrame containing the lines
     """
 
-    list_linestrings = [linestring]
+    df_l = lines.copy()  # can use lines directly without copying
```

Comment:
for the purpose of this function the columns that may be needed are bus0, bus1 and geometry beyond the index.
However, if we preserve as we should the index, we can always recover bus0 and bus1 from the original dataframe, so, here probably keeping geomerty and the index would be enough.
Probably, we could subset this dataframe as well to increase performances

Reply:
i have removed the sub setting for buses and lines for now.